src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+export interface Figure {
+  shape: `triangle` | `circle` | `rectangle`;
+  color: `red` | `green` | `blue`;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('the side should be bigger than 0');
+    }
+
+    if (!this.isValidTriangle(a, b, c)) {
+      throw new Error('a longest side should be bigger than sum of two others');
```

Comment:
The error message should provide a more detailed description. It could say 'The length of the longest side must be less than the sum of the other two sides to form a valid triangle.'